packages/react-ui-components/src/components/CodeEditor/CodeEditor.tsx

```diff
@@ -1,53 +1,45 @@
-import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import React, { useCallback, useEffect, useMemo } from 'react';
 import Editor, { useMonaco } from '@monaco-editor/react';
-import { DEFAULT_EDITOR_LANGUAGE, DEFAULT_EDITOR_THEME, DEFAULT_PADDING_TOP, FONT_NOT_FOUND_ERROR } from './config';
-import { ECodeEditorTheme, ICodeEditorProps, OnEditorChangeCallback } from './types';
+import { DEFAULT_CODE_EDITOR_LANGUAGE, DEFAULT_CODE_EDITOR_OPTIONS, DEFAULT_CODE_EDITOR_THEME, KELVIN_CODE_EDITOR_THEME } from './config';
+import { ECodeEditorTheme, ICodeEditorProps, OnCodeEditorChange } from './types';
 import { KvLoader } from '../stencil-generated';
-import { getEditorOptions } from './utils';
+import { getEditorOptions, getFontOptions } from './utils';
+import { useFontsApi } from '../../utils';
+
+const CodeEditorLoader = () => <KvLoader isLoading />;
 
 export const KvCodeEditor = ({
 	code,
-	readOnly = false,
-	loadingComponent = <KvLoader isLoading />,
-	language = DEFAULT_EDITOR_LANGUAGE,
-	theme = DEFAULT_EDITOR_THEME,
-	customTheme,
-	paddingTop = DEFAULT_PADDING_TOP,
+	language = DEFAULT_CODE_EDITOR_LANGUAGE,
+	theme = DEFAULT_CODE_EDITOR_THEME,
+	customTheme = KELVIN_CODE_EDITOR_THEME,
+	customOptions = DEFAULT_CODE_EDITOR_OPTIONS,
```

Comment:
Not sure if we should merge the provided custom options with the default ones. Since this object can be a bit extensive, probably we should not force the user to write every option and instead only define the ones he wants to override. 🤔

Reply:
my bad, left it there after making the changes that were discussed [here](https://github.com/kelvininc/ui-components/pull/277#discussion_r1158574157)